### PR TITLE
Feature/nick e adpsfc prepbufr

### PR DESCRIPTION
**Pull request by nicholasesposito**
This adds the ADPSFC prepbufr updated api to the GDASApp.

The two files added are:
    ~/ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py
    ~/parm/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.json

This is validated in [ioda-converters issue 1293 ](https://github.com/JCSDA-internal/ioda-converters/issues/1293)
I can open a JEDI-T20 issue for the validation if that is needed.

### CODE REVIEW

File: parm/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "data_format"      : "prepbufr",
+  "subsets"          : [ "ADPSFC" ],
+  "source"           : "prepBUFR",
+  "data_type"        : "ADPSFC",
+  "cycle_type"       : "{{ RUN }}",
+  "cycle_datetime"   : "{{ current_cycle | to_YMDH }}",
+  "dump_directory"   : "{{ DMPDIR }}",
+  "ioda_directory"   : "{{ COM_OBS }}",
+  "data_description" : "ADPSFC_prepbufr",
+  "data_provider"    : "U.S. NOAA/NESDIS"
```
Comment (CoryMartin-NOAA) [inline parm/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.json]: Not NESDIS, right?

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    logger.info(f" ... Executing QuerySet: get datatime: observation time ...")
+    # DateTime: seconds since Epoch time
+    # IODA has no support for numpy datetime arrays dtype=datetime64[s]
+    dhr = r.get('obsTimeMinusCycleTime', type='int64')
+
+    logger.info(f" ... Executing QuerySet: Done!")
+
+    logger.info(f" ... Executing QuerySet: Check BUFR variable generic \
+                dimension and type ...")
+    # Check BUFR variable generic dimension and type
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:142]: Do we need all of this printed for `info`? Can we make it `debug`?

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    # =========================
+    start_time = time.time()
+
+    logger.info(f"Creating derived variables - dateTime ...")
+
+    cycleTimeSinceEpoch = np.int64(calendar.timegm(time.strptime(
+                                   reference_time_full, '%Y%m%d%H%M')))
+    dateTime = Compute_dateTime(cycleTimeSinceEpoch, dhr)
+
+    logger.info(f"     Check derived variables type ... ")
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py]: Same comment about info vs debug

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        .write_data(zobqm)
+
+    # QualityMarker - Station Pressure
+    obsspace.create_var('QualityMarker/stationPressure', dtype=pobqm.dtype,
+                        fillval=pobqm.fill_value) \
+        .write_attr('long_name', 'Station Pressure Quality Marker') \
+        .write_data(pobqm)
+
+    # Station Elevation
+    obsspace.create_var('ObsValue/stationElevation', dtype=elv.dtype,
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:302]: is this an `ObsValue`?

File: parm/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.json (hunk 1 above)
Comment (nicholasesposito) [reply parm/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.json]: True. I'll update this

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py (hunk 2 above)
Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:142]: Yes, we can make it debug. I'll change that.

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py (hunk 3 above)
Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py]: I'll change all of them to debug

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py (hunk 4 above)
Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:302]: stationElevation (mnemonic elv) is an ObsValue while heightOfStation (mnemonic zob) is a MetaData

**Review by CoryMartin-NOAA — APPROVED** [review]

File: ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+        .write_attr('long_name', 'Station Pressure') \
+        .write_data(pob)
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logger.debug(f"Running time for splitting and output IODA: \
+                {running_time} seconds")
+
+    logger.debug("All Done!")
+
```
Comment (PraveenKumar-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:321]: @nicholasesposito I think you may need to remove one extra line from here for the coding norms.

Comment (PraveenKumar-NOAA) [reply ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py:321]: @nicholasesposito all checks are passing, its good then.

**Review by PraveenKumar-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**